### PR TITLE
Move from CircleCI to Github Actions

### PR DESCRIPTION
**Pull request by douglascamata**
Signed-off-by: Douglas Camata <159076+douglascamata@users.noreply.github.com>

This PR moves the build system from Circle CI to Github Actions.

### CODE REVIEW

File: .github/workflows/container.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: Container build & push jobs
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+*"
```
Comment (douglascamata) [inline .github/workflows/container.yaml:8]: Github's tag filtering capabilities are not using regex. So I use a simpler filter here to take anything of the shape `vX.Y.Z*` and later before pushing the release image the full regex from semver.org is applied and verified.

File: .github/workflows/container.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Cache for Docker's buildx
+        uses: actions/cache@v3
+        with:
+          path: .buildxcache/
+          key: ${{ runner.os }}-buildx-${{ hashFiles('**/*.go', 'Dockerfile', 'go.sum') }}
```
Comment (douglascamata) [inline .github/workflows/container.yaml:49]: The cache for Docker image building depends on changes on the source code, dockerfile itself, and dependencies. No cache usage is forced, we still rely on Docker's mechanisms to decide whether or not to use the cached layers available.

File: .github/workflows/container.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Snapshot container buid & push
+        run: make conditional-container-build-push
+
+      - name: Check semver tag
+        id: check-semver-tag
+        run: |
+          if [[ ${{ github.event.ref }} =~ ^refs/tags/v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$ ]]; then
```
Comment (douglascamata) [inline .github/workflows/container.yaml:60]: This regex comes out of https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

File: .github/workflows/container.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Login to Quay.io
+        uses: docker/login-action@v2
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
```
Comment (douglascamata) [inline .github/workflows/container.yaml:36]: Need maintainers help to set this up in Github.

**Comment by douglascamata** [comment]
We might need someone to “unregister” the project from CircleCI, because it still wants to build the project but doesn’t find a configuration file and then fails the check.

**Review by matej-g — APPROVED** [review]
Looks good after testing! :green_circle:

**Pull request merged**